.github/workflows/sites-test.yml

```diff
@@ -0,0 +1,250 @@
+    env:
+      REAL_AES_ENCRYPTION: ${{secrets.REAL_AES_ENCRYPTION}}
+      DRUPAL_DATABASE_NAME: drupal
+      DRUPAL_DATABASE_USERNAME: drupal
+      DRUPAL_DATABASE_PASSWORD: drupal
+      DRUPAL_DATABASE_HOST: mysql
+    services:
+      mysql:
+        image: mysql:5.7
+        env:
+          MYSQL_DATABASE: drupal
+          MYSQL_USER: drupal
+          MYSQL_PASSWORD: drupal
+          MYSQL_ROOT_PASSWORD: drupal
+        ports:
+          - 33306:3306
+        options: --health-cmd="mysqladmin ping" --health-interval=10s --health-timeout=5s --health-retries=3
+    steps:
+      - uses: actions/checkout@v4
+      - name: Restore Cache
+        uses: actions/cache@v4
+        with:
+          path: |
+            vendor
+            docroot/core
+            docroot/libraries
+            docroot/modules/contrib
+          key: 1.x-${{ hashFiles('blt/blt.yml') }}-${{ hashFiles('composer.json') }}-${{ hashFiles('composer.lock') }}
+          restore-keys: |
+            1.x-${{ hashFiles('blt/blt.yml') }}-${{ hashFiles('composer.json') }}-${{ hashFiles('composer.lock') }}
+            1.x-${{ hashFiles('blt/blt.yml') }}-${{ hashFiles('composer.json') }}-
+            1.x-
+      - run: git config --system --add safe.directory '*'
+      - name: Install Dependencies
+        run: |
+          mysql -h mysql -P 3306 -u root -pdrupal -e 'SET GLOBAL max_allowed_packet=67108864;' &&
+          rm -rf /var/www/html &&
+          ln -snf $GITHUB_WORKSPACE /var/www/html &&
+          composer install -n &&
+          vendor/bin/blt blt:telemetry:disable --no-interaction &&
+          blt settings &&
+          mkdir -p docroot/sites/default/files &&
+          chmod -R 777 docroot/sites/default/files/
+      - name: Codeception Tests
+        run: |
+          blt tests:phpunit:coverage --no-interaction
+  set_sites:
+    runs-on: ubuntu-latest
+    container:
+      image: pookmish/drupal8ci:latest
+    outputs:
+      matrix: ${{ steps.set-matrix.outputs.matrix }}
+    steps:
+      - uses: actions/checkout@v4
+      - name: Restore Cache
+        uses: actions/cache@v4
+        with:
+          path: |
+            vendor
+            docroot/core
+            docroot/libraries
+            docroot/modules/contrib
+          key: 1.x-${{ hashFiles('blt/blt.yml') }}-${{ hashFiles('composer.json') }}-${{ hashFiles('composer.lock') }}
+          restore-keys: |
+            1.x-${{ hashFiles('blt/blt.yml') }}-${{ hashFiles('composer.json') }}-${{ hashFiles('composer.lock') }}
+            1.x-${{ hashFiles('blt/blt.yml') }}-${{ hashFiles('composer.json') }}-
+            1.x-${{ hashFiles('blt/blt.yml') }}-
+      - id: set-matrix
+        run: |
+          composer install -n &&
+          vendor/bin/blt blt:telemetry:disable --no-interaction &&
+          SITES=$(echo -n "$(blt blt:config:get multisites | shuf -n 10)" | jq -R -s -c 'split("\n") | sort')
+          echo "matrix={\"site\": $SITES}" >> $GITHUB_OUTPUT
```

Comment:
Neat

Reply:
That was a tricky one to figure out too. But love how it works